org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/apt/model/ElementsImpl9.java

```diff
@@ -284,6 +288,37 @@ private JavaFileObject getSourceJavaFileObject(ReferenceContext referenceContext
 		return null;
 	}
 	@Override
+	public TypeElement getEnumConstantBody(VariableElement enumConstant) {
+		switch(enumConstant.getKind()) {
+			case ENUM_CONSTANT -> {
+				VariableElementImpl variableImpl = (VariableElementImpl) enumConstant;
+				if (variableImpl._binding instanceof FieldBinding field) {
+					FieldDeclaration sourceField = field.sourceField();
+					if (sourceField == null)
+						return null;
+					if (sourceField.initialization instanceof QualifiedAllocationExpression qualAlloc) {
+						TypeDeclaration decl = qualAlloc.anonymousType;
+						if (decl == null)
+							return null;
+
+						SourceTypeBinding declaringClass = (SourceTypeBinding) field.declaringClass;
+						TypeDeclaration enumTypeDecl = declaringClass.scope.referenceContext;
+						// The following code is copied from BlockScope#addAnonymousType() to force
+						// resolution of the local type binding ahead of a full CU#resolve()
```

Comment:
At this point, could we already have `enumTypeDecl.binding` set due to a previous call to the current method?